libbeat/processors/uuid/generator/elasticsearch/mac.go

```diff
@@ -0,0 +1,101 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package elasticsearch
+
+import (
+	"crypto/rand"
+	"net"
+)
+
+// Golang port of https://github.com/elastic/elasticsearch/blob/a666fb2266/server/src/main/java/org/elasticsearch/common/MacAddressProvider.java
+
+type id []byte
+
+const addrLen = 6
+
+func getSecureMungedMACAddress() ([]byte, error) {
+	addr, err := getMacAddress()
+	if err != nil {
+		return nil, err
+	}
+
+	if !isValidAddress(addr) {
+		addr, err = constructDummyMulticastAddress()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	munged := make([]byte, addrLen)
+	_, err = rand.Read(munged)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
the `crypt/rand.Read` can easily fail if there is not enough entropy in the system. Especially during system startup. In this case we read a few zeros.